### PR TITLE
Point3 mean to throw an exception when input size=0 .

### PR DESCRIPTION
**Pull request by Alexma3312**
https://github.com/borglab/gtsam/pull/464 follow up.
A simple fix to throw an exception when the input size equals to zero in the Point3 mean functions. @jlblancoc

### CODE REVIEW

File: gtsam/geometry/Point3.cpp [hunk 1]
```diff
@@ -77,6 +77,7 @@ double dot(const Point3 &p, const Point3 &q, OptionalJacobian<1, 3> H1,
 
 Point3Pair mean(const std::vector<Point3Pair> &abPointPairs) {
   const size_t n = abPointPairs.size();
+  if (n == 0) throw std::invalid_argument("input should have at least 1 pair of points");
```
Comment (varunagrawal) [inline gtsam/geometry/Point3.cpp:80]: Slightly more descriptive error message?

`Point3Pair::mean input vector is empty`

File: gtsam/geometry/Point3.h [hunk 2]
```diff
@@ -62,6 +62,7 @@ GTSAM_EXPORT double dot(const Point3& p, const Point3& q,
 /// mean
 template <class CONTAINER>
 GTSAM_EXPORT Point3 mean(const CONTAINER& points) {
+  if (points.size() == 0) throw std::invalid_argument("input should have at least 1 point");
```
Comment (varunagrawal) [inline gtsam/geometry/Point3.h:65]: Same as above.

**Review by varunagrawal — APPROVED** [review]
LGTM except for a minor comment.

File: gtsam/geometry/Point3.cpp (hunk 1 above)
Comment (Alexma3312) [reply gtsam/geometry/Point3.cpp:80]: sure

**Review by dellaert — APPROVED** [review]

**Pull request merged**